skimage/feature/_canny.py

```diff
@@ -155,126 +330,52 @@ def canny(image, sigma=1., low_threshold=None, high_threshold=None, mask=None,
     # because who knows what lies beyond the edge of the image?
     #
     check_nD(image, 2)
-    dtype_max = dtype_limits(image, clip_negative=False)[1]
 
     if low_threshold is None:
+        warnings.warn("Setting low_threshold to None is deprecated. "
+                      "It will raise an error starting from version 0.19. "
+                      "To remove this warning, use the default value or "
+                      "explicitely set its value.", FutureWarning)
         low_threshold = 0.1
-    else:
-        low_threshold = low_threshold / dtype_max
 
     if high_threshold is None:
+        warnings.warn("Setting high_threshold to None is deprecated. "
+                      "It will raise an error starting from version 0.19. "
+                      "To remove this warning, use the default value or "
+                      "explicitely set its value.", FutureWarning)
         high_threshold = 0.2
-    else:
-        high_threshold = high_threshold / dtype_max
 
-    if mask is None:
-        mask = np.ones(image.shape, dtype=bool)
+    if use_quantiles:
+        if preserve_range:
+            dtype_max = dtype_limits(image, clip_negative=False)[1]
+            low_threshold = low_threshold / dtype_max
+            high_threshold = high_threshold / dtype_max
+        if high_threshold < low_threshold:
+            raise ValueError(
+                "low_threshold should be lower then high_threshold")
+        if high_threshold > 1.0 or low_threshold > 1.0:
+            raise ValueError("Quantile thresholds must not be > 1.0")
+        if high_threshold < 0.0 or low_threshold < 0.0:
+            raise ValueError("Quantile thresholds must not be < 0.0")
 
-    def fsmooth(x):
-        return img_as_float(gaussian(x, sigma, mode='constant'))
```

Comment:
In the previous version, `preserve_range` was implicitly set to `False` as the default argument of `gaussian`